seedpod_ground_risk/core/plot_server.py

```diff
@@ -198,26 +198,45 @@ def compose_overlay_plot(self, x_range: Optional[Sequence[float]] = (-1.6, -1.2)
 
                     t0 = time()
                     self._progress_bar_callback(10)
+                    # TODO: This will give multiple data layers, these need to be able to fed into their relevent pathfinding layers
+                    for annlayer in self.annotation_layers:
+                        new_layer = FatalityRiskLayer('Fatality Risk', ac=annlayer.aircraft['name'])
+                        self.add_layer(new_layer)
+                    self.remove_duplicate_layers()
+                    self._progress_bar_callback(20)
                     self.generate_layers(bounds_poly, raster_shape)
                     self._progress_bar_callback(50)
-                    plot = Overlay([res[0] for res in self._generated_data_layers.values()])
+                    plt_lyr = list(self._generated_data_layers)[0]
+                    plot = Overlay([self._generated_data_layers[plt_lyr][0]])
                     print("Generated all layers in ", time() - t0)
                     if self.annotation_layers:
-                        plot = Overlay([res[0] for res in self._generated_data_layers.values()])
-                        raw_datas = [res[2] for res in self._generated_data_layers.values()]
-                        raster_indices = dict(Longitude=np.linspace(x_range[0], x_range[1], num=raster_shape[0]),
-                                              Latitude=np.linspace(y_range[0], y_range[1], num=raster_shape[1]))
-                        raster_grid = np.sum(
-                            [remove_raster_nans(res[1]) for res in self._generated_data_layers.values() if
-                             res[1] is not None],
-                            axis=0)
-                        raster_grid = np.flipud(raster_grid)
-                        raster_indices['Latitude'] = np.flip(raster_indices['Latitude'])
+                        plot = Overlay([self._generated_data_layers[plt_lyr][0]])
+                        res = []
+                        for dlayer in self.data_layers:
```

Comment:
Why are we iterating `self.data_layers` when we are iterating `self._generated_data_layers` inside this loop anyway?